doc/dev/matrix_solver.rst

```diff
@@ -0,0 +1,144 @@
+.. _matrix_solver:
+
+Matrix Solvers
+==============
+
+Cable Equation
+--------------
+
+At the heart of the time evolution step Arbor we find a linear system that must
+be solved once per time step. This system arises from the cable equation
+
+.. math::
+   C \partial_t V = \frac{\sigma}{2\pi r}\partial_x(r^2\partial_x V) + I
+
+   I: \mbox{External currents}
+
+   r: \mbox{cable radius}
+
+   \sigma: \mbox{conductivity}
+
+after discretisation into CV's, application of the FVM, and choosing an implicit
+Euler time step as
+
+.. math::
+   \left(\frac{\sigma_i C_i}{\Delta\,t} + \sum_{\delta(i, j)} a_{ij}\right)V_i^{k+1} - \sum_{\delta(i, j)} a_ij V_i^{k+1} = \frac{\sigma_i C_i}{\Delta\,t}V_i^k + \sigma_i I_i
+
+where :math:`\delta(i, j)` indicates whether two CVs are adjacent. It is written
+in form of a sparse matrix, symmetric by construction.
+
+Note that the cable equation is non-linear as the currents :math:`I` potentially
+depend on :math:`V`. We obtain the current value of :math:`I` by splitting the
+full operator into the 'matrix' (described here) and 'mechanism' parts (giving
+the currents) and assuming one part being a constant during the update of the
+respective other. However, for the matrix solver, we model _linear_ dependencies
+:math:`I = gV + J` and collect all higher orders in the pretended constant
+:math:`J` to improve accuracy and stability of the solver. This requires the
+computation of the symbolic derivative :math:`g = \partial_V I` during
+compilation of the mechanisms and an update to :math:`g(t)` when updating
+currents using that symbolic expression.
```

Comment:
I found this whole paragraph kind of confusing. It's a difficult (and important) section to explain so I get it. But maybe it could use a rewrite?

Reply:
Rewrite applied.